internal/controller/merge.go

```diff
@@ -213,12 +213,16 @@ func routersAreCompatible(r, toMerge *frr.RouterConfig) error {
 	return nil
 }
 
-// Verifies that two neighbors are compatible for merging, assuming they belong to the same router.
+// Verifies that two neighbors are compatible for merging.
 func neighborsAreCompatible(n1, n2 *frr.NeighborConfig) error {
-	if n1.Addr != n2.Addr {
+	if n1.Addr != "" && n1.Addr != n2.Addr {
 		return fmt.Errorf("neighbors with different addresses (%s != %s) are not compatible for merging", n1.Addr, n2.Addr)
 	}
 
+	if n1.Iface != "" && n1.Iface == n2.Iface {
```

Comment:
same (with the &&s), also why isn't this `iface != iface`? we merge when the neighbor is the same